### DISCUSSION TITLE
Translations get sent on each request

### DISCUSSION THREAD

**Discussion by wuifdesign** [i18next/next-i18next#1768] [Q&A] [answered]
why do the translations get sent on each request? i have a namespace "common" and on each rout change the same translations get sent to the user making each rout change cost multiple kb for sending same data every time. 

isn't this producing a lot of traffic and delay for users? couldn't this be cached client side?

**Comment by isaachinman** [comment, marked as answer]
Please see https://github.com/isaachinman/next-i18next/issues/1215.

This question of "shared props" is a generic issue that NextJs faces, and is not specific to `next-i18next`.

**Reply by wuifdesign** [reply]
in don't mind adding it to each page. what i mean is that i want to exclude it from the response sent to the user.

i see the problem in the overhead of loading all translations on each route change. if the common translations have 1 MB (because there are many translations) each route change has to load the 1MB file again, and on slow networks it will take 10sec. but in reality it doesn't need any of this as all of the translations are already loaded anyway.

it will slow down user experience.

is the only way to use i18next (and not next-i18next) and load translations there, as they will only load once? With the tradeoff of not having it on SSR.

something i would expect is to load the translations via props on the initial request, but then it loades only the missing namespaces via ajax on route change (and don't get all of them again in the serverprops).

**Reply by isaachinman** [reply]
@wuifdesign Can you check [this](https://github.com/locize/next-i18next-locize/blob/main/next-i18next.config.js) example out and see if the localStorage cache might help you?

**Reply by wuifdesign** [reply]
this will not change the server behavior, or am i wrong? it just adds loaders to browser, but on serverside all translations will be added to the request anyway.
